File: python/allergies/allergies.py
class Allergies(object):
    allergens = [
        'eggs',
        'peanuts',
        'shellfish',
        'strawberries',
        'tomatoes',
        'chocolate',
        'pollen',
        'cats',
    ]

    def __init__(self, score):
        self.allergies = [
            allergen
            for i, allergen in enumerate(self.allergens)
            if (score & (1 << i)) != 0
        ]

    def is_allergic_to(self, item):
        return item in self.allergies

    @property
    def lst(self):
        return self.allergies

File: nim/allergies/allergies.nim
from strutils import parseEnum

type 
  Allergen = enum
    eggs
    peanuts
    shellfish
    strawberries
    tomatoes
    chocolate
    pollen
    cats

  Allergies* = object 
    score*: int


proc isAllergicTo(allergies: Allergies, allergen: Allergen): bool =
  (allergies.score and (1 shl ord(allergen))) != 0

proc isAllergicTo*(allergies: Allergies, allergen: string): bool =
  allergies.isAllergicTo(parseEnum[Allergen](allergen))


proc lst*(allergies: Allergies): seq[string] =
  for allergen in Allergen:
    if allergies.isAllergicTo(allergen):
      result.add($allergen)
